implementations/tutorial_coherent_vqls.py

```diff
@@ -0,0 +1,541 @@
+7. If the outcome of the measurement is the ground state, the system collapses to
+   :math:`|\Psi\rangle :=  A |x\rangle/\sqrt{\langle x |A^\dagger A |x\rangle}`.
+   If the outcome is not the ground state, the experiment should be repeated.
+
+
+Estimating the cost function
+>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+
+From a technical point of view, the previous steps represent the most difficult part of the algorithm. 
+Once we have at our disposal the quantum system prepared in the state :math:`|\Psi\rangle`,
+it is very easy to compute the cost function.
+Indeed one could simply continue the previous protocol with the following two steps:
+
+8. Apply :math:`U_b^\dagger` to the system.
+9. Measure the system in the computational basis. The probability of finding it
+   in the ground state (given the ancillary qubits measured in their ground state),
+   is :math:`|\langle 0 | U_b^\dagger |\Psi \rangle|^2 = |\langle b | \Psi \rangle|^2`.
+
+So, with sufficiently many shots of the previous experiment, one can directly estimate
+the cost function of the problem.
+
+Importantly, the operations of steps 7 and 8 commute. Therefore all the measurements can be
+delayed at the end of the quantum circuit as represented in the figure at the top of this tutorial,
+making the structure of the experiment more straightforward.  
+
+A simple example
+^^^^^^^^^^^^^^^^
+
+In this tutorial we apply the previous theory to the following simple example 
+based on a system of 3 qubits, which was already considered in Ref. [1] and also reproduced in PannyLane (VQLS):
+
+.. math::
+        \begin{align}
+        A  &=  c_0 A_0 + c_1 A_1 + c_2 A_2 = \mathbb{I} + 0.2 X_0 Z_1 + 0.2 X_0, \\
+        \\
+        |b\rangle &= U_b |0 \rangle = H_0  H_1  H_2 |0\rangle,
+        \end{align}
+
+where :math:`Z_j, X_j, H_j` represent the Pauli :math:`Z`, Pauli :math:`X` and Hadamard gates applied to the qubit with index :math:`j`.
+
+This problem is computationally quite easy since a single layer of local rotations is enough to generate the
+solution state, i.e., we can use the following simple ansatz:
+
+.. math::
+        |x\rangle = V(w) |0\rangle = \Big [  R_y(w_0) \otimes  R_y(w_1) \otimes  R_y(w_2) \Big ]  H_0  H_1  H_2 |0\rangle.
+
+
+In the code presented below we solve this particular problem, by following the general scheme of the CVQLS previously discussed.
+Eventually we will compare the quantum solution with the classical one.
+
+General setup
+-------------
+This Python code requires *PennyLane* and the plotting library *matplotlib*.
+
+"""
+
+import pennylane as qml
+from pennylane import numpy as np
+import matplotlib.pyplot as plt
+
+##############################################################################
+# Setting of the main hyper-parameters of the model
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+n_qubits = 3                # Number of system qubits.
+m = 2                       # Number of ancillary qubits
```

Comment:
```suggestion
m = 2                       # Number of ancillary qubits.
```
If you're gonna go with periods, you missed one here :)

Reply:
Removed all of them :-)